Chess/painter.cpp

```diff
@@ -1,25 +1,105 @@
-#include "painter.h"
-
-Painter::Painter() {}
-
-void Painter::initializeWindow() {
-
-    initwindow(WINDOW_WIDTH, WINDOW_HEIGHT, "Chess");
-}
-
-void Painter::initializeBoard() {
-    // TODO
-}
-
-void Painter::initializePieces() {
-    initializeBlackPieces();
-    initializeWhitePieces();
-}
-
-void Painter::initializeBlackPieces() {
-    // TODO
-}
-
-void Painter::initializeWhitePieces() {
-    // TODO
-}
+#include "painter.h"
+
+Painter::Painter() {}
+
+void Painter::initializeWindow()
+{
+
+    initwindow(WINDOW_WIDTH, WINDOW_HEIGHT, "Chess");
+}
+
+void Painter::initializeBoard()
+{
+
+    for(int i = 1; i <= 8; i++)
+    {
+        for(int j = 1; j <= 8; j++)
+        {
+
+            if((i + j) % 2)
+                drawSquare(i, j, 0, "Backgrounds/Black.jpg");
+            else
+                drawSquare(i, j, 1, "Backgrounds/White.jpg");
+
+        }
+    }
+}
+
+void Painter::initializePieces()
+{
+    initializeBlackPieces();
+    initializeWhitePieces();
+}
+
+void Painter::initializeBlackPieces()
+{
+
+    for(int i = 1; i <= 8; i++)
+    {
+        for(int j = 1; j <= 2; j++)
+        {
+            if((i + j) % 2)
+            {
+                if(j == 1)
+                    loadPiece(i, j, BlackOnBlackPaths[i / 2 - 1]);
+
+                else if(j == 2)
+                    loadPiece(i, j, BlackOnBlackPaths[4]);
+            }
+            else
+            {
+                if(j == 1)
+                    loadPiece(i, j, BlackOnWhitePaths[(8 - i) / 2]);
+
+                else if(j == 2)
+                    loadPiece(i, j, BlackOnWhite[4]);
+            }
+        }
+    }
+
+}
+
+void Painter::initializeWhitePieces()
+{
+    // TODO
+}
+
+void Painter::drawSquare(int row, int column, int patrat, char image[])
+{
+    int diffCol = (column - 1) * SQUARE_SIZE;
+    int diffLin = (row - 1 ) * SQUARE_SIZE;
+
+    line(OFFSET_X + diffLin, OFFSET_Y + diffCol,
+         OFFSET_X + diffLin, OFFSET_Y + diffCol + SQUARE_SIZE);
+
+    line(OFFSET_X + diffLin, OFFSET_Y + diffCol + SQUARE_SIZE,
+         OFFSET_X + diffLin + SQUARE_SIZE, OFFSET_Y + diffCol + SQUARE_SIZE);
+
+    line(OFFSET_X + diffLin + SQUARE_SIZE, OFFSET_Y + diffCol + SQUARE_SIZE,
+         OFFSET_X + diffLin + SQUARE_SIZE, OFFSET_Y + diffCol);
+
+    line(OFFSET_X + diffLin + SQUARE_SIZE, OFFSET_Y + diffCol,
+         OFFSET_X + diffLin, OFFSET_Y + diffCol);
+
+    if(patrat == 0)

```

Comment:
Sa pastram consistenta si sa punem totul in engleza
Totodata, am putea sa denumim altfel variabila asta ca sa fie cat mai clean posibil: sa punem in config un enum pentru culori (o sa ne trebuiasca si in alte contexte probabil) care sa aiba doua valori: BLACK si WHITE. Si aici sa pasezi un astfel de enum in loc de int, ca sa fie mai usor de citit, iar variabila sa se denumeasca color si if-ul ar fi ceva de genul:
```if(color == BLACK)```